src/App/mermaidData/syncApiDataIntoOfflineStorage/SyncApiDataIntoOfflineStorage.js

```diff
@@ -6,6 +6,54 @@ const SyncApiDataIntoOfflineStorage = class {
 
   _dexiePerUserDataInstance
 
+  _getTableNamesWhereSyncingWasRejectedListedPerProject =
+    function _getTableNamesWhereSyncingWasRejectedListedPerProject(pushResponse) {
+      const projectsWithSyncErrors = {}
+
+      try {
+        const responseData = pushResponse.data
+
+        const mermaidApiSyncTableNames = Object.keys(responseData)
+
+        mermaidApiSyncTableNames.forEach((apiSyncTableName) => {
+          responseData[apiSyncTableName].forEach((itemSyncResponseObject) => {
+            const isSync403 = itemSyncResponseObject.status_code === 403
+            const syncErrorProjectName = itemSyncResponseObject.data.project_name
+            const syncErrorProjectId = itemSyncResponseObject.data.project_id
+
+            // to differentiate from other 403s, the proxy metric here is that if the
+            // response includes a project name, this means a user in not allowed to sync
+            const isUserDeniedFromSyncingToProject = isSync403 && syncErrorProjectName
+
+            if (isUserDeniedFromSyncingToProject) {
+              const projectsOtherApiDataTablesThatRejectedSyncing =
+                projectsWithSyncErrors[syncErrorProjectId]?.apiDataTablesThatRejectedSyncing ?? []
+
+              projectsWithSyncErrors[syncErrorProjectId] = {
+                name: syncErrorProjectName,
+                apiDataTablesThatRejectedSyncing: [
+                  ...projectsOtherApiDataTablesThatRejectedSyncing,
+                  apiSyncTableName,
+                ],
+              }
+            }
+          })
+        })
+
+        // the format of the returned object will be
+        // { projectId: { name: string, apiDataTablesThatRejectedSyncing: string[] } }
+        return projectsWithSyncErrors
+      } catch (error) {
+        console.error('Not able to assess syncing success', error)
+
+        return {}
```

Comment:
I dont love swallowing errors, but if the api changes and the above Object.keys isnt fed an object, the app would crash  which is overkill for a syncing error.... I might look into a different approach 

My hedge here is that we are using console.error, which is useless once we build, but Im thinking we can ctrl-F-all to replace them with a logger service if we want to be more serious about error handling.

Reply:
which reminds me about error documentation - I feel like we should be able to expect consistent error formatting, feels shaky